src/files/files_client.py

```diff
@@ -377,74 +384,60 @@ def _remote_file_exists(self, path: str) -> bool:
             # If there was an error getting metadata, the file probably doesn't exist
             return False
 
-        """
-    Folder creation is yet supported by back end well.
-
     def create_folder(self, path: str) -> bool:
-        "
-        Create a folder in the remote storage.        
+        """
+        Create a folder in the remote storage.
         In case of S3, should create a zero-byte object with a trailing slash and special
-        content type to represent a folder in object storage.        
+        content type to represent a folder in object storage.
         Args:
-            path: Remote path where to create the folder. Must end with '/'            
+            path: Remote path where to create the folder. Must end with '/'
         Returns:
             True if folder creation was successful, False otherwise
-        "
-        if not path.endswith('/'):
+        """
+        if not path.endswith("/"):
             error_msg = f"Folder path must end with '/': {path}"
             self.logger.error(error_msg)
             raise ValueError(error_msg)
-            
+
         remote_path = self._extract_path_from_url(path)
-        
+
         try:
-            # Create an empty bytes buffer as a file-like object
-            empty_buffer = io.BytesIO(b"")
-            
-            # Create a File object with the empty buffer
-            empty_file = File(
-                payload=empty_buffer,
-                file_name="",
-                #mime_type="application/x-directory"
-                mime_type = "application/octet-stream"
+            # TBD: ideally mime-type created would be "application/x-directory"
+            call_args = genapi._get_put_object_kwargs(
+                orgFriendlyId=self.organization_id,
+                filePath=remote_path,
+                # file=file_obj,
             )
-            
-            # Create an empty object with folder content type
-            response = put_object.sync_detailed(
-                org_friendly_id=self.organization_id,
-                file_path=remote_path,
-                client=self.client,
-                # content_type="application/x-directory",  # Set in File object instead
-                body=PutObjectBody(file=empty_file)
-            )
-            
+            response = self.client.httpx_request_and_process(call_args)
+
             if response.status_code == 200:
                 return True
             else:
-                self.logger.error(f"Failed to create folder {path}: Status code {response.status_code}")
+                self.logger.error(
+                    f"Failed to create folder {path}: Status code {response.status_code}"
+                )
```

Comment:
if the user asks to create a folder, and folder creation fails, we should raise an exception